Document connection pool sizes and their overrides (#541)
digital-asset/docs.daml.com: docs/2.8.0/docs/canton/usermanual/persistence.rst

::

    canton.participants.<participant-name>.parameters.ledger-api-server-parameters.indexer.ingestion-parallelism = Y

The number ``Z`` of the connections used by the exclusive sequencer writer component is the final parameter that can be controlled.

::

    canton.sequencers.<sequencer-name>.sequencer.high-availability.exclusive-storage.max-connections = Z

A Canton participant node will establish up to ``X + Y + 2`` permanent connections with the database, whereas a domain node
will use up to ``X`` permanent connections, except for a sequencer with HA setup that will allocate up to ``2X`` connections. During
startup, the node will use an additional set of at most ``X`` temporary connections during database initialisation.

The number ``X`` represents an upper bound of permanent connections and is divided internally for different purposes,
depending on the implementation. Consequently, the actual size of the write connection pool, for example, could be smaller.

> Some of the allotted connections will be taken by the *read* pool, some by the *write* pool and a single additional connection

Reviewer: ```suggestion
Some of the allotted connections will be taken by the *read* pool, some will be taken by the *write* pool, and a single additional connection
```

+------------------+--------------------+--------------------+--------------------+

The results of the divisions are always rounded down unless they yield a zero. In that case, a minimal pool
size of 1 is ascertained.

The values obtained from that formula can be overridden using explicit configuration settings for the *ledger api* ``A``,
the *read* ``R``, the *write* ``W`` pools.

::

    canton.participants.<participant-name>.storage.parameters.connection-allocation.num-reads = R-overwrite
    canton.participants.<participant-name>.storage.parameters.connection-allocation.num-writes = W-overwrite
    canton.participants.<participant-name>.storage.parameters.connection-allocation.num-ledger-api = A-overwrite

Similar parameters exist also for other Canton node types:

::

    canton.sequencers.sequencer.storage.parameters.connection-allocation...
    canton.mediators.mediator.storage.parameters.connection-allocation...
    canton.domain-managers.domain_manager.storage.parameters.connection-allocation...

Where a node operates a combined *read/write* connection pool, the numbers for ``R`` and ``W`` overwrites are added
together to determine the overall pool size.

> The effective connection pool sizes are reported by the Canton nodes at start-up

Reviewer: ```suggestion
The effective connection pool sizes are reported by the Canton nodes at startup.
```

::

    canton.participants.<participant-name>.parameters.ledger-api-server-parameters.indexer.ingestion-parallelism = Y

A Canton participant node will establish up to ``X + Y + 2`` permanent connections with the database, whereas a domain node
will use up to ``X`` permanent connections, except for a sequencer with HA setup that will allocate up to ``2X`` connections. During
startup, the node will use an additional set of at most ``X`` temporary connections during database initialisation.

The number ``X`` represents an upper bound of permanent connections and is divided internally for different purposes,
depending on the implementation. Consequently, the actual size of the write connection pool, for example, could be smaller.

> Following table summarizes the detailed split of the connection pools in different canton nodes. ``R`` signifies a *read* pool, ``W``

Reviewer: ```suggestion
The following table summarizes the detailed split of the connection pools in different Canton nodes. ``R`` signifies a *read* pool, ``W``
```

> a *write* pool and ``A`` a *ledger api* pool, ``RW`` a combined *read/write* pool and ``EW`` an *exclusive writer* pool.

Reviewer: ```suggestion
a *write* pool , ``A`` a *ledger api* pool, ``RW`` a combined *read/write* pool, and ``EW`` an *exclusive writer* pool.
```

|                  | EW = 1             |                    |
+--------------+-------------------+--------------------+
| Mediator         | R = X / 2          |                    |
|                  | W = X / 2 - 1      |                    |
|                  | EW = 1             |                    |
+--------------+-------------------+--------------------+
| Sequencer        | RW =  X            | column 3           |
+--------------+-------------------+--------------------+
| Sequencer writer | R = X / 2          |                    |
|                  | W = X / 2 - 1      |                    |
|                  | EW = 1             |                    |
+--------------+-------------------+--------------------+
| Sequencer        | R = Z / 2          | column 3           |
| exclusive writer | W = Z / 2          | column 3           |
+--------------+-------------------+--------------------+
| Domain manager   | R = X / 2          |                    |
|                  | W = X / 2 - 1      |                    |
|                  | EW = 1             |                    |
+--------------+-------------------+--------------------+

> The results of the divisions are always rounded down, unless the result yields a zero. In that

Reviewer: ```suggestion
The results of the divisions are always rounded down unless the result yields a zero. In that
```

> case a minimal pool size of 1 is ascertained.

Reviewer: ```suggestion
case, a minimal pool size of 1 is ascertained.
```

> The sizes of the

Reviewer: wondering if there needs to be more to this sentence?

Author: It was a wip, more is coming
